### PR TITLE
Dump statementInfo into local file that the format is parquet

### PR DESCRIPTION
**Pull request by zhang1002**
### _Why are the changes needed?_
We need to dump data from mem into file regularly.
This PR is just for local file, the file's format is parquet.

Notice:
1. There will store lots of small files if we don't use data-lake
2. Need to discuss how to set partition for this schema
3. If the schema is changed, how to query the old data

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/latest/tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: externals/kyuubi-spark-monitor/src/main/scala/org/apache/kyuubi/engine/spark/monitor/KyuubiStatementMonitor.scala [hunk 1]
```diff
@@ -69,22 +98,38 @@ object KyuubiStatementMonitor extends Logging{
...
-  private def removeAndDumpStatementInfoFromQueue(): Unit = {
-    // TODO: Just for test
-    kyuubiStatementQueue.clear()
+  private def dumpStatementInfoFromQueue(): Unit = kyuubiStatementQueue.synchronized {
+    var statementInfoList = new util.ArrayList[KyuubiStatementInfo]()
+    val size = kyuubiStatementQueue.size()
+    (0 to size-1).foreach { _ =>
+      statementInfoList.add(kyuubiStatementQueue.poll())
+    }
+    if (DataFrameUtil.getStructType(classOf[KyuubiStatementInfo]).nonEmpty) {
```
Comment (yaooqinn) [inline externals/kyuubi-spark-monitor/src/main/scala/org/apache/kyuubi/engine/spark/monitor/KyuubiStatementMonitor.scala:118]: org.apache.spark.sql.Encoders.product[xxx].schema?

**Review by yaooqinn — APPROVED** [review]

File: externals/kyuubi-spark-monitor/src/main/scala/org/apache/kyuubi/engine/spark/monitor/KyuubiStatementMonitor.scala [hunk 2]
```diff
@@ -17,27 +17,59 @@
...
+        "file://" + System.getProperty("user.dir") + "/dump_data"
+      }
+    case dir: String =>
+      "file://" + dir
+  }
+
+  /**
+   * The max size that trigger to dump data is 0.7*maxCapacity
+   */
+  private val maxSize: Int = scala.math.floor(0.7*maxCapacity).intValue()
```
Comment (ulysses-you) [inline externals/kyuubi-spark-monitor/src/main/scala/org/apache/kyuubi/engine/spark/monitor/KyuubiStatementMonitor.scala:57]: `maxStatementSizeInMemory`

File: externals/kyuubi-spark-monitor/src/main/scala/org/apache/kyuubi/engine/spark/monitor/KyuubiStatementMonitor.scala [hunk 3]
```diff
@@ -69,22 +101,40 @@ object KyuubiStatementMonitor extends Logging{
...
+  private def dumpStatementInfoFromQueue(): Unit = kyuubiStatementQueue.synchronized {
+    var statementInfoList: List[KyuubiStatementInfo] = List()
+    val size = kyuubiStatementQueue.size()
+    (0 to size-1).foreach { _ =>
+      statementInfoList = kyuubiStatementQueue.poll() :: statementInfoList
+    }
+    var schema = classToStructTyeMap.get(classOf[KyuubiStatementInfo])
+    if (schema == null) {
+      schema = ScalaReflection.schemaFor[KyuubiStatementInfo].dataType.asInstanceOf[StructType]
+      classToStructTyeMap.put(classOf[KyuubiStatementInfo], schema)
```
Comment (ulysses-you) [inline externals/kyuubi-spark-monitor/src/main/scala/org/apache/kyuubi/engine/spark/monitor/KyuubiStatementMonitor.scala:128]: can we move this code as static code block ?

File: externals/kyuubi-spark-monitor/src/main/scala/org/apache/kyuubi/engine/spark/monitor/KyuubiStatementMonitor.scala [hunk 4]
```diff
@@ -69,22 +101,40 @@ object KyuubiStatementMonitor extends Logging{
...
+    }
+    val rowList = new java.util.ArrayList[Row]()
+    for (kyuubiStatementInfo <- statementInfoList) {
+      rowList.add(Row.fromTuple(kyuubiStatementInfo))
+    }
+    sparkSession.createDataFrame(rowList, schema)
+      .write.mode(SaveMode.Append)
+      .format("parquet")
+      .save(dumpLocalDir + "/statement_info.parquet")
```
Comment (ulysses-you) [inline externals/kyuubi-spark-monitor/src/main/scala/org/apache/kyuubi/engine/spark/monitor/KyuubiStatementMonitor.scala:137]: this param is a path not a file name

File: externals/kyuubi-spark-monitor/src/main/scala/org/apache/kyuubi/engine/spark/monitor/KyuubiStatementMonitor.scala [hunk 5]
```diff
@@ -17,27 +17,59 @@
...
+   */
+  private val maxSize: Int = scala.math.floor(0.7*maxCapacity).intValue()
+
+  if (dumpEnable) {
+    val dumpDataByTimeThresholdThreadPool = new ScheduledThreadPoolExecutor(1)
+    dumpDataByTimeThresholdThreadPool.scheduleWithFixedDelay(new Runnable {
+      override def run(): Unit = {
+        dumpStatementInfoFromQueue()
+      }
+    }, 60, timeThreshold, TimeUnit.SECONDS)
```
Comment (ulysses-you) [inline externals/kyuubi-spark-monitor/src/main/scala/org/apache/kyuubi/engine/spark/monitor/KyuubiStatementMonitor.scala:65]: `}, timeThreshold, timeThreshold, TimeUnit.SECONDS)`

**Pull request closed without merging**